lib/util/sourceMap.js

```diff
@@ -0,0 +1,58 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Furkan Erdem @log101
+*/
+
+"use strict";
+
+const { SourceMapConsumer } = require("source-map");
+
+/** @typedef {import("../Compilation").ModuleCallback} ModuleCallback */
+/** @typedef {import("../Dependency").RealDependencyLocation} RealDependencyLocation */
+/** @typedef {import("../WebpackError")} WebpackError */
+
+/**
+ * Updates error location (line number and column) using sourcemap
+ *
+ * @param {WebpackError} err The error with the misaligned location.
+ * @param {ModuleCallback} callback callback
+ * @returns {Promise<void>}
+ */
+const updateErrorLocation = async (err, callback) => {
+	let consumer;
+	try {
+		consumer = await new SourceMapConsumer(err.module.originalSource().map());
```

Comment:
I see, I think we need to solve it better, without extra deps, because we already have such logic in webpack-source - https://github.com/webpack/webpack-sources/tree/main/lib/helpers, let's send a PR to there and make API required for this public API, because we will need it for https://github.com/webpack-contrib/terser-webpack-plugin/blob/master/src/index.js#L261 too and other plugins it, I found useful if we will make it everywhere and will use less deps

Reply:
Wow, I totally missed that! Alright I'll be sending it there. But what do you mean by making it required?

Reply:
I mean now it is not public API so let's extract it from our logic and make it public, I just chose the wrong word, a lot of answers :smile:

Reply:
Alright .)